R/readGAINS2025final.R

```diff
@@ -0,0 +1,285 @@
+      rename(EMF30_MIXED = EMF30_AGG) %>%
+      filter(!(EMF30_MIXED %in% sectors_to_ignore))
+    # Read in selected detailed sectors (sectors_det_selected)
+    df_sel_det_sectors <- read.csv(paste0(filepath_beginning, "_det_", filepath_ending)) %>%
+      rename(EMF30_MIXED = EMF30_DET) %>%
+      filter(EMF30_MIXED %in% sectors_det_selected)
+    # Combine
+    df_sectors <- rbind(df_agg_sectors, df_sel_det_sectors)
+    return(df_sectors)
+  }
+
+  # Function to correct selected sectors (sectors_to_correct)
+  correctSectors <- function(df_uncorrected, filepath_ending) {
+    # Read in correct data
+    df_NatGas <- read.csv(paste0("Transformations_NatGas_", filepath_ending)) %>%
+      rename(EMF30_MIXED = EMF30_AGG)
+    df_Coal <- read.csv(paste0("Transformations_Coal_", filepath_ending)) %>%
+      rename(EMF30_MIXED = EMF30_AGG)
+    # Remove sectors to correct from uncorrected dataframe
+    df_uncorrected <- df_uncorrected %>% filter(!(EMF30_MIXED %in% sectors_to_correct))
+    # Add correct data
+    df_corrected <- rbind(df_uncorrected, df_Coal, df_NatGas)
+  }
+
+  # ================================================================================================
+  # AIR POLLUTANT NAMES ============================================================================
+  # ================================================================================================
+
+  # Function to maps from GAINS2025 to REMIND (oldGAINS) air pollutant names
+  fixPolNames <- function(mag) {
+    polnamesmap <- c(
+      "CO" = "CO",
+      "NOx" = "NOX",
+      "BC" = "PM_BC",
+      "OC" = "PM_OC",
+      "SO2" = "SO2",
+      "NH3" = "NH3",
+      "VOC" = "VOC"
+    )
+    mag <- mag[, , polnamesmap]
+    getItems(mag, "species") <- names(polnamesmap)
+    return(mag)
+  }
+
+  if (subtype == "emifacs") {
+    # ================================================================================================
+    # EMISSION FACTORS ===============================================================================
+    # Read emission factors for historical timesteps, ScenarioMIP scenarios, CLE, SLE and MTFR =======
+    # Combine aggregated sectors and selected detailed sectors (Waste) ===============================
+    # Remove or correct erroneous data according to information from GAINS team ======================
+    # ================================================================================================
+
+    # HISTORICAL =====================================================================================
+    histefs <- readSectors("emission_factors", "hist_final_2025-07-02.csv")
+    histefs <- correctSectors(histefs, "hist-2025-07-04-CORRECTION.csv")
+
+    # Convert to long format and standardize column names
+    histefs <- pivot_longer(histefs, 7:length(names(histefs)), names_prefix = "X", names_to = "year") %>%
+      select(-timestamp) %>%
```

Comment:
To avoid the linter warning "no symbol found in scope", use `select(-"timestamp")`for dplyr::select